### PR TITLE
Add multi threading

### PR DESCRIPTION
**Pull request by danila19991**
Added multithreading to compute unit.

### CODE REVIEW

**Comment by danila19991** [comment]
Some comments to current realisation. By some experiments was discovered that construction and destruction of QTcpSocket must be performed in thread, where it would be used. That's why was created a special class worker(for executing function in anouther thread). With it in constructor and destructor code which refers to QTcpSocket executes in thread, where it would be used.

File: CMakeLists.txt [hunk 1]
```diff
@@ -27,6 +27,7 @@ find_package(Threads)
 find_package(Qt5 COMPONENTS Core Network REQUIRED)
 
 # Include sub-projects.
+add_subdirectory ("worker")
```
Comment (ptrdiff) [inline CMakeLists.txt]: Change name to more understandable one.

File: Executor/Executor.cpp [hunk 2]
```diff
@@ -22,7 +61,7 @@ void Executor::slotMoveRobot(double j1, double j2, double j3, double j4, double
     {
         speed = (abs(_currentCoords[0] - j1) + abs(_currentCoords[1] - j2) + 
             abs(_currentCoords[2] - j3) + abs(_currentCoords[3] - j4) + 
-            abs(_currentCoords[4] - j5) + abs(_currentCoords[5] - j6))/20;
+            abs(_currentCoords[4] - j5) + abs(_currentCoords[5] - j6))/30;
```
Comment (ptrdiff) [inline Executor/Executor.cpp]: Empirical const.

File: Executor/Executor.h [hunk 3]
```diff
@@ -1,7 +1,11 @@
 #ifndef EXECUTOR_H
 #define EXECUTOR_H
 
+#include "../FanucAdapter/FanucAdapter.h"
+#include "../RCAConnector/RCAConnector.h"
+
 #include <QObject>
+#include <qthread.h>
```
Comment (ptrdiff) [inline Executor/Executor.h]: Use qt-standart header.

File: Executor/Executor.h [hunk 4]
```diff
@@ -11,10 +15,18 @@ class Executor : public QObject
...
-    Executor(QObject *parent = nullptr);
+    Executor(std::string, int, int, QObject *parent = nullptr);
+
+    void shutDown();
+
+    void launch();
 
 signals:
 
+    void startRCA();
```
Comment (ptrdiff) [inline Executor/Executor.h]: Remove signals to non-public field.

File: FanucAdapter/FanucAdapter.cpp [hunk 5]
```diff
@@ -13,38 +11,71 @@ FanucAdapter::FanucAdapter(std::string serverIP, int port, QObject* parent):
 QObject(parent),
 _prevData(""),
 _serverPort(static_cast<quint16>(port)),
-_socket(std::make_unique<QTcpSocket>(this)),
-_serverIP(std::move(serverIP))
+_serverIP(serverIP)
 {
-    connect(_socket.get(), &QTcpSocket::readyRead, this, &FanucAdapter::slotReadFromServer);
+    this->moveToThread(&_myThread);
+    auto func = [this]() {
```
Comment (ptrdiff) [inline FanucAdapter/FanucAdapter.cpp]: Temporary variable.

File: FanucAdapter/FanucAdapter.cpp [hunk 6]
```diff
@@ -13,38 +11,71 @@ FanucAdapter::FanucAdapter(std::string serverIP, int port, QObject* parent):
 QObject(parent),
 _prevData(""),
```
Comment (ptrdiff) [inline FanucAdapter/FanucAdapter.cpp]: Initialize all fields.

File: FanucAdapter/FanucAdapter.cpp [hunk 7]
```diff
@@ -13,38 +11,71 @@ FanucAdapter::FanucAdapter(std::string serverIP, int port, QObject* parent):
...
+    
     for (int i = 0; i < 3; ++i) {
         if (TryConnect(10000))
         {
             flag = true;
             break;
         }
         std::cout << "Can't create first connect to server" << std::endl;
-    }
+    }//*/
```
Comment (ptrdiff) [inline FanucAdapter/FanucAdapter.cpp]: Delete.

File: RCAConnector/RCAConnector.h [hunk 8]
```diff
@@ -42,6 +52,12 @@ private slots:
 
 protected:
 
+    Worker                          _initialiser;
+
+    bool                            _isInitialiesd{ false };
```
Comment (ptrdiff) [inline RCAConnector/RCAConnector.h]: Delete.

**Pull request merged**